java-checks/src/main/java/org/sonar/java/checks/MembersDifferOnlyByCapitalizationCheck.java

```diff
@@ -171,13 +172,18 @@ private static String getSymbolTypeName(Symbol symbol) {
   private static List<Symbol> extractMembers(Symbol.TypeSymbol classSymbol, boolean ignorePrivate) {
     List<Symbol> results = Lists.newLinkedList();
     for (Symbol symbol : classSymbol.memberSymbols()) {
-      if ((symbol.isVariableSymbol() || methodButNotConstructor(symbol)) && !(symbol.isPrivate() && ignorePrivate)) {
+      if ((variableButNotHiddenMember(symbol) || methodButNotConstructor(symbol)) && !(symbol.isPrivate() && ignorePrivate)) {
         results.add(symbol);
       }
     }
     return results;
   }
 
+  private static boolean variableButNotHiddenMember(Symbol symbol) {
+    String name = symbol.name();
+    return symbol.isVariableSymbol() && !"this".equals(name) && !"super".equals(name);
```

Comment:
Several things : 
-  as you handle the "this" symbol, it should be explicitely tested
- it should be explicitely mentioned in jira ticket
- This is not hidden members... implicit eventually but hidden has a total different meaning in java semantic : http://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.4.8.2